packages/test/test-end-to-end-tests/src/test/opPerfTelemetry.spec.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	ITestObjectProvider,
+	ITestContainerConfig,
+	DataObjectFactoryType,
+	ITestFluidObject,
+	createTestConfigProvider,
+} from "@fluidframework/test-utils";
+import { type SharedString } from "@fluidframework/sequence";
+import { describeCompat, itExpects } from "@fluid-private/test-version-utils";
+import { ILoaderProps } from "@fluidframework/container-loader";
+
+/**
+ * Regression tests for issue where clients loading a container with a write connection (vs the read connection
+ * that became the default) would never generate OpRoundtripTime telemetry.
+ */
+describeCompat(
+	"All clients generate OpRoundtripTime telemetry",
+	// TODO: this should be "2.0.0-rc.2.0.0" (I think) once the bugs with describeCompat are fixed and versions are used correctly
+	"NoCompat",
+	(getTestObjectProvider, apis) => {
+		const { SharedString } = apis.dds;
+
+		const customConfigProvider = createTestConfigProvider();
+		customConfigProvider.set("Fluid.Container.ForceWriteConnection", true);
+		const loaderPropsThatForceWriteConnection: Partial<ILoaderProps> = {
+			configProvider: customConfigProvider,
+		};
+
+		const stringId = "stringKey";
+		const configWithReadConnection: ITestContainerConfig = {
+			fluidDataObjectType: DataObjectFactoryType.Test,
+			registry: [[stringId, SharedString.getFactory()]],
+		};
+		const configWithWriteConnection = {
```

Comment:
Is the write connection config just an extension of the read connection config with the loader props?

Reply:
Yep. I basically wanted to use the exact same configuration, with the only difference that the initial connection to the service should be established in write mode instead of the default of read.